cli/src/main/java/com/devonfw/tools/ide/environment/EnvironmentVariablesPropertiesFile.java

```diff
@@ -104,27 +104,34 @@ public void save() {
       this.context.info("Converting legacy properties to {}", newPropertiesFilePath);
     }
     List<VariableLine> lines = new ArrayList<>();
-    // TODO FixMe: we should simply skip reading if not exists instead of creating an empty file and then reading it. 
-    //             Also we should do mkdirs on parent folder to ensure it exists - see Review from https://github.com/devonfw/IDEasy/pull/374
-    if (!Files.exists(newPropertiesFilePath)) {
-      try {
-        Files.createFile(newPropertiesFilePath);
-      } catch (IOException e) {
-        throw new IllegalStateException("Failed to create properties file with" + newPropertiesFilePath, e);
+
+    // Ensure the parent directory exists
+    try {
+      if (newPropertiesFilePath.getParent() != null) {
+        Files.createDirectories(newPropertiesFilePath.getParent());
       }
-    }
-    try (BufferedReader reader = Files.newBufferedReader(newPropertiesFilePath)) {
-      String line;
-      do {
-        line = reader.readLine();
-        if (line != null) {
-          VariableLine variableLine = VariableLine.of(line, this.context, getSource());
-          lines.add(variableLine);
-        }
-      } while (line != null);
     } catch (IOException e) {
-      throw new IllegalStateException("Failed to load existing properties from " + this.propertiesFilePath, e);
+      throw new IllegalStateException("Failed to create parent directories for " + newPropertiesFilePath, e);
+    }
```

Comment:
You have access to `context` and `FileAccess` here. We invented `FileAccess` to avoid repeating this low-level stuff all the time:
```suggestion
  Path parentFolder = newPropertiesFilePath.getParent();
  if (parentFolder != null) {
    this.context.getFileAccess().mkdirs(parentFolder);
  }
```

Reply:
Just for the record: Under which circumstances will `newPropertiesFilePath.getParent()` return `null`?
IMHO this can never happen in reality.
However, I just learned that NIO `Path` API is not so well designed since you cannot traverse up to the parent of a relative path in case you "leave" that relative path. I would not have expected this:
```
Path srcMain = Path.of("src/main");
Path src = srcMain.getParent(); // same as Path.of("src") what is fine
Path basedir = src.getParent(); // not pointing to the current working directory but giving null instead
```

Therefore you instead need to do
```
Path srcMain = Path.of("src/main").toAbsolutePath();
```
to make it work as expected...

So very safe that you check for `null`. One could somehow set it as `Path.of("ide.properties")` in case we once support `ide.properties` in `IDE_HOME` and that could cause an NPE.
Anyway I ensure that all this should really never happen here:
https://github.com/devonfw/IDEasy/blob/1e62bd2a4a949f9875cac16118bf5108940eb431/cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java#L143

Reply:
All solved. Now 👍